lib_eio/eio.mli

```diff
@@ -223,6 +227,9 @@ module Stdenv : sig
       To use this, see {!Domain_manager}.
   *)
 
+  val process : <process : #Process.mgr as 'a; ..> -> 'a
```

Comment:
Yep done in https://github.com/ocaml-multicore/eio/pull/330/commits/1d0f3def3087405fb55968a0089e22c1f3221be4 although now there's a slight discrepancy with having `Eio.Process` instead of `Eio.Process_mgr` but I thought it would be better to keep `Eio.Process` like `Lwt_process`.